File: app/src/main/java/org/runnerup/features/StartUiState.java
package org.runnerup.features;

/** Start-screen view updates (GPS/HR/Wear, start button). */
final class StartUiState {

  private final StartFragment fragment;
  private final StartGpsController gps;
  private final StartHrController hr;

  StartUiState(StartFragment fragment, StartGpsController gps, StartHrController hr) {
    this.fragment = fragment;
    this.gps = gps;
    this.hr = hr;
  }

  void updateView() {
    if (fragment.getView() == null) {
      return;
    }
    fragment.performUpdateNewStartButton();
    gps.updateGpsView();
    hr.updateHrIndicator();
    gps.updateSatelliteInfo();
    boolean hrPresent = hr.updateHrView();
    boolean wearPresent = fragment.performUpdateWearOsView();
    fragment.performUpdateNoDevicesConnected(!hrPresent && !wearPresent);
  }
}

File: app/src/main/kotlin/org/runnerup/features/StartUiState.kt
package org.runnerup.features

/** Start-screen view updates (GPS/HR/Wear, start button). */
internal class StartUiState(
    private val fragment: StartFragment,
    private val gps: StartGpsController,
    private val hr: StartHrController,
) {

  fun updateView() {
    if (fragment.view == null) {
      return
    }
    fragment.performUpdateNewStartButton()
    gps.updateGpsView()
    hr.updateHrIndicator()
    gps.updateSatelliteInfo()
    val hrPresent = hr.updateHrView()
    val wearPresent = fragment.performUpdateWearOsView()
    fragment.performUpdateNoDevicesConnected(!hrPresent && !wearPresent)
  }
}
